django/contrib/sitemaps/__init__.py

```diff
@@ -122,6 +124,13 @@ def location(self, item):
 
     def get_protocol(self, protocol=None):
         # Determine protocol
+        if not any((self.protocol, protocol)):
+            warnings.warn(
+                "http as the default sitemap protocol is deprecated. "
+                "https will be the new default protocol. "
+                "Set an explicit 'protocol' to silence this warning.",
+                category=RemovedInDjango41Warning
+            )
```

Comment:
I think the wording could be improved a little here:

```suggestion
        if not self.protocol and not protocol:
            warnings.warn(
                "The default protocol will be changed from 'http' to 'https' "
                "in Django 5.0. Explicitly pass 'protocol' to this method or "
                "to the constructor to silence this warning.",
                category=RemovedInDjango50Warning,
            )
```

Don't forget to update the tests too.